client/js/lounge.js

```diff
@@ -310,10 +312,60 @@ $(function() {
 		return msg;
 	}
 
-	function buildChannelMessages(channel, messages) {
-		return messages.reduce(function(docFragment, message) {
-			docFragment.append(buildChatMessage({
-				chan: channel,
+	function updateCondensedText(condensed, type) {
+		var obj = {};
+		var types = ["part", "quit", "join", "nick", "mode"];
+
+		for (var i in types) {
+			var msgType = types[i];
+			obj[msgType] = condensed.data(msgType) || 0;
+			if (type === msgType) {
+				obj[msgType]++;
+				condensed.data(msgType, obj[msgType]);
+			}
+		}
+
+		var text = "";
+
+		for (var j in types) {
+			var messageType = types[j];
+			if (obj[messageType]) {
+				text += text === "" ? "" : ", ";
+				text += obj[messageType] + " " + messageType;
+				if (messageType === "nick" || messageType === "mode") {
+					text += " change";
+				}
+				text += obj[messageType] > 1 ? "s" : "";
+			}
+		}
+		$(condensed.children(".text")[0]).text(text);
+	}
+
+	function appendMessage(container, chan, chanType, messageType, msg) {
+		if (["join", "part", "quit", "nick", "mode"].indexOf(messageType) !== -1 && chanType !== "lobby") {
+			var lastChild = container.children("div.msg").last();
+			if (lastChild && $(lastChild).hasClass("condensed") && !$(msg).hasClass("message")) {
+				lastChild.append(msg);
+				updateCondensedText(lastChild, messageType);
+			} else if (lastChild && $(lastChild).is(".join, .part, .quit, .nick, .mode")) {
+				var condensed = buildChatMessage({msg: {type: "condensed", time: msg.attr("data-time")}, chan: chan});
+				condensed.append(lastChild);
+				updateCondensedText(condensed, lastChild.attr("data-type"));
+				condensed.append(msg);
+				container.append(condensed);
+				updateCondensedText(condensed, messageType);
```

Comment:
Ah yes. The reason is that the updateCondensedText takes in a message and updates using that. So, here we are adding 2 messages (because we have 2 messages to put in the condensed section because we are waiting until we have 2 messages before condensing). I'll think up a nicer way to do this.